domain/user/state/state.go

```diff
@@ -723,24 +743,23 @@ WHERE user_uuid = $M.uuid
 // isRemoved returns the value of the removed field for the user with the
 // supplied uuid. If the user does not exist an error that satisfies
 // usererrors.NotFound will be returned.
-func (st *State) isRemoved(ctx context.Context, tx *sqlair.TX, uuid user.UUID) (bool, error) {
+func (st *State) isRemoved(ctx context.Context, tx *sqlair.TX, name string) (bool, error) {
```

Comment:
These convenience methods are non-idiomatic from the other state layers. Essentially, we're creating a new prepare for _every_ retry of the query. This is overkill. Instead, we need to come up with another pattern.

 - ensureUserAuthentication
 - setPasswordHash
 - removePasswordHash
 - setActivationKey
 - removeActivationKey
 - isRemoved

Reply:
Yep. I can reorganise this too.